src/line_launcher/mod.rs

```diff
@@ -0,0 +1,100 @@
+use midir::MidiOutputConnection;
+use std::sync::mpsc::Receiver;
+use wmidi::{Channel, MidiMessage, Note, Velocity};
+
+use crate::BeatNumber;
+
+enum State {
+    NotPlaying,
+    Playing { next_note_index: usize },
+    WaitingToFireFinalNoteOff,
+}
+
+const CHANNEL: Channel = Channel::Ch1;
+const VELOCITY: u8 = 100;
+
+pub struct LineLauncher {
+    beat_message_receiver: Receiver<BeatNumber>,
+    state: State,
+    notes: Vec<u8>,
+    output: MidiOutputConnection,
+}
+
+impl LineLauncher {
+    pub fn new(beat_message_receiver: Receiver<BeatNumber>, output: MidiOutputConnection) -> Self {
+        Self {
+            beat_message_receiver,
+            state: State::NotPlaying,
+            notes: vec![60, 53, 55, 58, 60, 61, 63, 65, 64],
+            output,
+        }
+    }
+
+    pub fn listen(&mut self) -> () {
+        loop {
+            let beat_message = self.beat_message_receiver.recv().unwrap();
+            match self.state {
+                State::NotPlaying
+                    if beat_message.quarter_note == 1 && beat_message.sixteenth_note == 1 =>
```

Comment:
Would it make sense for this condition to be exposed as a function on `beat_message`?

Reply:
Sure, exposed as `beat_message.is_beginning_of_measure()`